crates/engine/tree/src/tree/invalid_block_hook.rs

```diff
@@ -42,3 +42,19 @@ impl<N: NodePrimitives> InvalidBlockHook<N> for InvalidBlockHooks<N> {
         }
     }
 }
+
+impl<N: NodePrimitives> InvalidBlockHooks<N> {
+    pub fn on_invalid_block_with_cache_check(
+        &self,
+        invalid_header_cache: &InvalidHeaderCache,
+        parent_header: &SealedHeader<N::BlockHeader>,
+        block: &RecoveredBlock<N::Block>,
+        output: &BlockExecutionOutput<N::Receipt>,
+        trie_updates: Option<(&TrieUpdates, B256)>,
+    ) {
+        if invalid_header_cache.get(&block.hash()).is_some() {
+            return;
+        }
+        self.on_invalid_block(parent_header, block, output, trie_updates);
```

Comment:
this is correct, but we want this as a fn inside this type so that


https://github.com/paradigmxyz/reth/blob/bffdda43129c207deeccf6ec2f0f30b884da5e45/crates/engine/tree/src/tree/mod.rs#L2399-L2399

becomes `fn on_invalid_block` with the function body you have